### PR TITLE
Dockerfile cleanup

### PR DESCRIPTION
**Pull request by shleeable**
Sister PR for the Flodgatt PR. #13286 - I've just refactored, and applied some "best practices" as raised by the lint tool.

Upgrade to Ruby 2.6.6

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,117 +1,119 @@
...
-	apt -y install wget python && \
+	dpkgArch="$(dpkg --print-architecture)" && \
+	case "${dpkgArch##*-}" in \
+		amd64) ARCH='x64';; \
+		ppc64el) ARCH='ppc64le';; \
+		s390x) ARCH='s390x';; \
+		arm64) ARCH='arm64';; \
+		armhf) ARCH='armv7l';; \
+		i386) ARCH='x86';; \
+		*) echo "unsupported architecture"; exit 1 ;; \
```
Comment (ThibG) [inline Dockerfile]: Erroring out on unsupported architectures was the first thing the Dockerfile did. Now, it'll do a bunch of stuff before that. I think it should still error out as soon as possible.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (shleeable) [reply Dockerfile]: Please test again.

File: Dockerfile [hunk 2]
```diff
@@ -1,117 +1,119 @@
...
-	  --with-shared \
-	  --disable-install-doc && \
-	ln -s /opt/jemalloc/lib/* /usr/lib/ && \
-	make -j$(nproc) > /dev/null && \
-	make install
-
-ENV PATH="${PATH}:/opt/ruby/bin:/opt/node/bin"
-
+RUN apt-get --no-install-recommends -y install build-essential bison libyaml-dev libgdbm-dev libreadline-dev \
+    libncurses5-dev libffi-dev zlib1g-dev libssl-dev && \
```
Comment (SuperSandro2000) [inline Dockerfile]:
```suggestion
RUN apt-get --no-install-recommends -y install \
    build-essential \
    bison \
    libgdbm-dev \
    libncurses5-dev \
    libreadline-dev \
    libffi-dev \
    libssl-dev \
    libyaml-dev \
    zlib1g-dev \    
  && \
```

**Comment by SuperSandro2000** [comment]
and ``&&`` should be at the beginning of the line so that you can add and remove lines without touching the lines above or below.

**Pull request closed without merging**